### PR TITLE
test(heading): add cypress tests

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour
Refactor for heading component to use new cypress-component-react framework for testing.
I have added test.js and deleted steps.js and feature file.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- [x]  Run npx cypress open-ct to check if there is newly added test.file
- [x]  Check if the heading.test.js file passed
- [x]  Compare all tests were moved/refactored from .feature file -> .test.js file
- [x] Check if all unnecessary files were deleted, including .json file

### CODE REVIEW

File: cypress/locators/heading/index.js [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
+  SUBHEADER_PREVIEW,
+  DIVIDER_PREVIEW,
+  PILL_PREVIEW,
+} from "./locators";
 
 // component preview locators
 export const headingPreview = () => cy.get(HEADING_PREVIEW);
 export const subheaderPreview = () => cy.get(SUBHEADER_PREVIEW);
 export const headingTitle = () => headingPreview().find("div > h1");
+export const pillPreview = () => cy.get(PILL_PREVIEW);
```
Comment (DlgSHi) [inline cypress/locators/heading/index.js:12]: we already have the `pill` declared :)

File: cypress/locators/heading/locators.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
 // component preview locators
 export const HEADING_PREVIEW = 'div[data-component="heading"]';
 export const SUBHEADER_PREVIEW = 'div[data-element="subtitle"]';
+export const DIVIDER_PREVIEW = '[data-element="divider"]';
+export const PILL_PREVIEW = 'span[data-component="pill"]';
```
Comment (DlgSHi) [inline cypress/locators/heading/locators.js:5]: we have it already defined  here -> `cypress/locators/pill/index.js`

**Review by DlgSHi — CHANGES_REQUESTED** [review]

**Review by DlgSHi — CHANGES_REQUESTED** [review]
Also you need to delete `heading.feature` and `heading.json`

File: src/components/heading/heading.test.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check pill for Heading component",
+      (pillText) => {
+        CypressMountWithProviders(
+          <HeadingComponent pills={<Pill>{pillText}</Pill>} />
+        );
+        getComponent("pill").should("have.text", pillText);
```
Comment (DlgSHi) [inline src/components/heading/heading.test.js:118]: we already have it defined in file I sent you :) could we just import it from the `pill -> index.js` file ?

**Review by DlgSHi — DISMISSED** [review]

File: cypress/locators/heading/index.js [hunk 4]
```diff
@@ -1,6 +1,12 @@
...
+  SUBHEADER_PREVIEW,
+  DIVIDER_PREVIEW,
+} from "./locators";
 
 // component preview locators
 export const headingPreview = () => cy.get(HEADING_PREVIEW);
 export const subheaderPreview = () => cy.get(SUBHEADER_PREVIEW);
 export const headingTitle = () => headingPreview().find("div > h1");
+export const dividerPreview = () => cy.get(DIVIDER_PREVIEW);
+export const sepratorPreview = () => cy.get(HEADING_PREVIEW).find("div > hr");
```
Comment (Parsium) [inline cypress/locators/heading/index.js:12]:
```suggestion
export const separatorPreview = () => cy.get(HEADING_PREVIEW).find("div > hr");
```

File: src/components/heading/heading.test.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
+import * as React from "react";
+import Heading from "./heading.component";
+import Pill from "../../components/pill";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+import {
+  headingPreview,
+  headingTitle,
+  sepratorPreview,
```
Comment (Parsium) [inline src/components/heading/heading.test.js:9]:
```suggestion
  separatorPreview,
```

**Review by Parsium — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**